### ISSUE TITLE
Hardcoded HKEY_LOCAL_MACHINE, unable to retrieve system env variables on Windows

### ISSUE THREAD

**Issue by darkmastermindz** [jiri-otoupal/pycrosskit#35]
**Describe the bug**

Pycrosskit hardcodes winreg values in `windows.py` to `HKEY_CURRENT_USER`. However, I'm looking to override that value `HKEY_CURRENT_SYSTEM` to get access to system environment variables. Without digging through the source code to override `windows.py`, this results in retrieving those variables from user env rather than system env as it errors out in `__get_policy_key_readonly` function.

**To Reproduce**
Setting the following reg_path to get values when retrieving from `SysEnv().get(key,reg_path, registry)` will result in error

```python
reg_path = r'SYSTEM\CurrentControlSet\Control\Session Manager\Environment'
SysEnv().get(key="Path", reg_path=reg_path, registry=True)
```

Snippet from `windows.py` example of hardcoding
```python
    @classmethod
    def __get_policy_key_readonly(cls, reg_path: str):

        root = winreg.ConnectRegistry(None, winreg.HKEY_CURRENT_USER)
        policy_key = winreg.OpenKeyEx(root, reg_path)
        return policy_key, root
```

**Expected behavior**

I'm trying to use Pycross kit to get system environment variables instead of using the winreg implementation which works as follows:

```python
import winreg

reg_path = r'SYSTEM\CurrentControlSet\Control\Session Manager\Environment'

reg_key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, reg_path)

system_environment_variables = winreg.QueryValueEx(reg_key, 'Path')[0]
```
 propose to change `windows.py` to the following to allow override in values for reg_key and set `winreg.HKEY_CURRENT_USER` as default to not change any current behavior
 
```python
    @classmethod
    def __get_policy_key_readonly(cls, reg_path: str, reg_key = winreg.HKEY_CURRENT_USER):

        root = winreg.ConnectRegistry(None, reg_key)
        policy_key = winreg.OpenKeyEx(root, reg_path)
        return policy_key, root
```

```python
 @classmethod
    def set(
            cls,
            key,
            value,
            subkey="",
            reg_path=r"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall",
            reg_key=winreg.HKEY_CURRENT_USER,
            registry=True,
            silent=False,
    ):
        """
        Set Environment Variable
        :param silent: If permission error should fail silently
        :param key: Variable Name
        :param value: Variable Value
        :param subkey: Sub-Key under key ( like file in folder )
        :param reg_path: Register path for windows
        :param reg_key: WinReg register key for windows
        :param registry: Only for windows, if true variable is set to registry path
                if false, variable is set to environment variables
        """
        if registry:
            root = winreg.ConnectRegistry(None, reg_key)


```

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Version: 22H2

**Comment by darkmastermindz** [comment]
Hi @jiri-otoupal, appreciate this package and excited to use it in my Python 3.11 project! 
I want to offer this code suggestion and PR #36 for review.

**Comment by jiri-otoupal** [comment]
I had merged PR #36 thank you for contribution !
